### ISSUE TITLE
Sonar scan fails if there are no class files available

### ISSUE THREAD

**Issue by vara-prasad** [SonarQubeCommunity/sonar-findbugs#46]
Sonar scan fails during findbugs execution in the case where there are java source files available and there are no class files available.
In some cases, the class files might be not be available and for this case, findbugs execution should be skipped and should not be failed.
Earlier in sonar-findbugs-plugin-3.2, findbugs execution gets skipped and now we are seeing the exception being raised and sonar scan fails.
Any specific reason on why the behaviour is changed ?
Can we skip the findbugs execution (fallback to old logic) instead of making the scan to fail.

**Comment by h3xstream** [comment]
> Sonar scan fails during findbugs execution in the case where there are java source files available and there are no class files available.

Please give more details about your project. Why are there no class? Are some Java files compiled or none?

**Comment by pcuttle** [comment]
We also have this problem. We run a standard sonarqube on all jars we create via maven, and have resource only jars. These resource only jars have package-info.java  file so that we get _some_ size and code coverage metrics in order that the sonarqube dashboard will show the charts that fail to render if the size and code coverage indications are missing.

This was working quite happily in 3.4.2 and is now broken in 3.4.3, also having the side effect failing the automated builds running on our build server. It may seem like a nice change to make, but the change in behaviour means that we have to either special case these jars manually, or revert to 3.4.2 and never update the plugin

**Comment by vara-prasad** [comment]
@h3xstream 
In some cases we will not have compiled classes available for java files. In this case we does not want to fail the sonar analysis and continue with the WARNING message from findbugs.

**Comment by rjbrandt** [comment]
We also have this issue, there are times when we want a scan but haven't built the code yet. 
I'm unclear as to whether this is going to be fixed in the next release, as it is we have to stay back at V3.3

**Comment by h3xstream** [comment]
@vara-prasad @rjbrandt As design, if you want to voluntary skip FindBugs analysis explicitly create a profile with no rule. #37 

@pcuttle Yup. This edge case should be cover by. #36

I will post an update here when a release package is create.

**Comment by vara-prasad** [comment]
@h3xstream We are using multi module project and we use same profile for all the modules.
This happens for one of the sub modules. 
The suggestion provided by you is not feasible with project setup we have.
Can we have this as an enhancement to log the message instead of failing the analysis by throwing exception.

**Comment by fjakop** [comment]
We have the same problem as @pcuttle, everything was fine now the modules without java code fail our entire build. We do not consider this an edge case.

**Comment by jtognazzi** [comment]
We also have this error with a multi module project.
How can I see the list of module without class file ?

**Comment by h3xstream** [comment]
#56 was create to add an optional option to skip the warnings.

**Comment by jrobledog** [comment]
We are experiencing the same issue. We have the same profile for all projects and the fact is that our moultimodule project sometimes would not compile but anyway we want to analyse the source code by skipping somehow the bytecode analysis from findbugs.
In previous versions of sonar runner we could use "sonar.excludePlugins" property to skip plugins, but now this property is deprecated and it seems that has no effect on analysis.

**Comment by olaoluphilip** [comment]
We are also experiencing this same issue on sonarqube 6.0. It works when sonar is ran from maven locally built fails on Bamboo

**Comment by kunfoo** [comment]
So what is the name of the option?

**Comment by Timunas** [comment]
I'm having a similar issue with sonar. I tried to get more logging information but using -X flags makes sonar not even parse projects.

I noticed that the project that is making sonar plugin to fail is one that doesn't build any artifact. Could this be the issue?

`16:39:36.396 16:39:36.387 INFO  - -------------  Scan awesome-lib
16:39:36.397 16:39:36.394 INFO  - Language is forced to scala
16:39:36.400 16:39:36.395 INFO  - Excluded sources for coverage: 
16:39:36.401 16:39:36.395 INFO  -   **/*Spec.scala
16:39:36.402 16:39:36.398 INFO  - Base dir: /home/jenkins/workspace/
16:39:36.402 16:39:36.398 INFO  - Working dir: /home/jenkins/workspace/
16:39:36.402 16:39:36.398 INFO  - Source paths: src/main/scala
16:39:36.403 16:39:36.399 INFO  - Source encoding: UTF-8, default locale: en_US
16:39:36.405 16:39:36.399 INFO  - Index files
16:39:36.406 16:39:36.399 INFO  - Excluded sources: 
16:39:36.406 16:39:36.399 INFO  -   **/*Spec.scala
16:39:36.406 16:39:36.399 INFO  - Excluded tests: 
16:39:36.406 16:39:36.399 INFO  -   **/*Spec.scala
16:39:36.406 16:39:36.401 INFO  - 1 files indexed
16:39:36.406 16:39:36.401 INFO  - 0 files ignored because of inclusion/exclusion patterns
16:39:36.407 16:39:36.405 INFO  - Quality profile for scala: Awesome
16:39:36.419 16:39:36.416 INFO  - Project is pulse-messaging (pulse-messaging)
16:39:36.425 16:39:36.416 INFO  - Project modules: []
16:39:36.425 16:39:36.416 INFO  - Report [/home/jenkins/workspace/target/jacoco.exec] exists?[false] isFile?[false]
16:39:36.427 16:39:36.416 INFO  - ScalaCoverageSensor: will not executed.
16:39:36.431 16:39:36.416 INFO  - 	foundReport = false
16:39:36.431 16:39:36.416 INFO  - 	hasScala    = true
16:39:36.431 16:39:36.416 INFO  - 	languages are [scala]
16:39:36.431 16:39:36.416 INFO  - ScalaItCoverageSensor: will not executed.
16:39:36.431 16:39:36.416 INFO  - ScalaItCoverageSensor: will not executed.
16:39:36.431 16:39:36.417 INFO  - Scala detected ... will execute ScalaSurefireSensor 
16:39:36.431 16:39:36.417 INFO  - JaCoCoSensor: JaCoCo report not found : /home/jenkins/workspace/target/jacoco.exec
16:39:36.431 16:39:36.417 INFO  - JaCoCoItSensor: JaCoCo IT report not found: /home/jenkins/workspace/target/jacoco-it.exec
16:39:36.431 16:39:36.418 INFO  - Sensor BaseMetricsSensor
16:39:36.431 16:39:36.426 INFO  - Sensor BaseMetricsSensor (done) | time=8ms
16:39:36.431 16:39:36.426 INFO  - Sensor Lines Sensor
16:39:36.431 16:39:36.426 INFO  - Sensor Lines Sensor (done) | time=0ms
16:39:36.432 16:39:36.426 INFO  - Sensor QProfileSensor
16:39:36.432 16:39:36.426 INFO  - Sensor QProfileSensor (done) | time=0ms
16:39:36.432 16:39:36.426 INFO  - Sensor ScoverageSensor
16:39:36.501 INFO: ------------------------------------------------------------------------
16:39:36.545 INFO: EXECUTION FAILURE
16:39:36.547 INFO: ------------------------------------------------------------------------
16:39:36.552 Total time: 9.774s
16:39:36.882 Final Memory: 130M/1299M
16:39:36.882 INFO: ------------------------------------------------------------------------
16:39:36.884 ERROR: Error during Sonar runner execution
16:39:36.885 ERROR: Unable to execute Sonar
16:39:36.891 ERROR: Caused by: None.get
16:39:36.893 ERROR: 
16:39:36.895 ERROR: To see the full stack trace of the errors, re-run SonarQube Runner with the -e switch.
16:39:37.015 ERROR: Re-run SonarQube Runner using the -X switch to enable full debug logging.
16:39:37.050 Build step 'Build using sbt' changed build result to FAILURE
16:39:37.184 Build step 'Build using sbt' marked build as failure`
